qpc/cred/test_openshift_cred_edit.py

```diff
@@ -0,0 +1,164 @@
+        sys.argv = [
+            "/bin/qpc",
+            "cred",
+            "edit",
+            "--name",
+            "openshift_cred",
+            "--token",
+        ]
+        CLI().main()
+        out, err = capsys.readouterr()
+        assert out == (
+            messages.OPENSHIFT_TOKEN
+            + "\n"
+            + messages.CRED_UPDATED % "openshift_cred"
+            + "\n"
+        )
+        assert err == ""
+
+    def test_edit_green_path(
+        self,
+        capsys,
+        requests_mock,
+        openshift_token_input,
+    ):
+        """Test openshift edit cred successfully."""
+        url_get = get_server_location() + CREDENTIAL_URI
+        url_patch = get_server_location() + CREDENTIAL_URI + "1/"
+        results = [
+            {
+                "id": 1,
+                "name": "openshift_cred",
+                "cred_type": OPENSHIFT_CRED_TYPE,
+                "token": "********",
+            }
+        ]
+        data = {"count": 1, "results": results}
+        requests_mock.get(url_get, status_code=200, json=data)
+        requests_mock.patch(url_patch, status_code=200)
+        sys.argv = [
+            "/bin/qpc",
+            "cred",
+            "edit",
+            "--name",
+            "openshift_cred_2",
+            "--token",
+        ]
+        CLI().main()
+        out, err = capsys.readouterr()
+        assert out == (
+            messages.OPENSHIFT_TOKEN
+            + "\n"
+            + messages.CRED_UPDATED % "openshift_cred_2"
+            + "\n"
+        )
+        assert err == ""
+
+    @pytest.mark.parametrize(
+        "status_code,err_log_message",
+        [
+            (401, "qpc server login"),
+            (500, "An internal server error occurred."),
+        ],
+    )
+    def test_edit_returning_error(  # pylint: disable=too-many-arguments
+        self,
+        capsys,
+        requests_mock,
+        openshift_token_input,
+        status_code,
+        err_log_message,
+    ):
+        """Test openshift edit cred w/ errors."""
+        url_get = get_server_location() + CREDENTIAL_URI
+        requests_mock.get(url_get, status_code=status_code, json=None)
+        sys.argv = [
+            "/bin/qpc",
+            "cred",
+            "edit",
+            "--name",
+            "openshift_cred_2",
+            "--token",
+        ]
+        with pytest.raises(SystemExit):
+            CLI().main()
+        out, err = capsys.readouterr()
+        assert out == ""
+        assert err_log_message in err
+
+    def test_edit_incorrect_arg(
+        self,
+        capsys,
+        requests_mock,
+    ):
+        """Test openshift edit cred with incorrect args."""
+        url_get = get_server_location() + CREDENTIAL_URI
+        url_patch = get_server_location() + CREDENTIAL_URI + "1/"
+        results = [
+            {
+                "id": 1,
+                "name": "openshift_cred",
+                "cred_type": OPENSHIFT_CRED_TYPE,
```

Comment:
What should happen if there was an attempt to change `type` to something else in an edit request? Should we test for that case?

Reply:
An error will be raised. I can add the test for this scenario.